src/cli/test_command.zig

```diff
@@ -1329,11 +1168,22 @@ pub const TestCommand = struct {
                     (Environment.isWindows and (strings.startsWith(arg, ".\\") or
                         strings.startsWith(arg, "..\\")))) break true;
             } else false) {
-                // One of the files is a filepath. Instead of treating the arguments as filters, treat them as filepaths
-                for (ctx.positionals[1..]) |arg| {
-                    results.appendAssumeCapacity(PathString.init(arg));
+                // One of the files is a filepath. Instead of treating the
+                // arguments as filters, treat them as filepaths
+                const file_or_dirnames = ctx.positionals[1..];
+                for (file_or_dirnames) |arg| {
+                    scanner.scan(arg) catch |err| switch (err) {
+                        error.OutOfMemory => bun.outOfMemory(),
+                        // don't error if multiple are passed; one might fail
+                        // but the others may not
+                        error.DoesNotExist => if (file_or_dirnames.len == 1) {
+                            Output.prettyErrorln("<red>The following filter did not match any test files:<r> {s}", .{arg});
```

Comment:
Let's make this formatting better

```suggestion
                            Output.errGeneric("Test filter did not match any files: <b>{}<r>", .{bun.fmt.quote(arg)});
```

<img width="570" alt="image" src="https://github.com/user-attachments/assets/b7285bed-9bb0-4251-8b3b-5fd6193ece81" />